static/app/components/events/viewHierarchy/index.tsx

```diff
@@ -101,6 +98,26 @@ function ViewHierarchy({viewHierarchy, project}: ViewHierarchyProps) {
     );
   };
 
+  const onScrollToNode = useCallback((node, scrollContainer, coordinates) => {
+    if (node) {
+      // When a user keyboard navigates to a node that's rendered in the "overscroll"
+      const lastCell = node.ref?.lastChild as HTMLElement | null | undefined;
+      if (lastCell) {
+        lastCell.scrollIntoView({
+          block: 'nearest',
+          inline: 'nearest',
+        });
+      }
+    } else if (coordinates) {
+      // When a user clicks on a wireframe node that's not rendered in the "overscroll"
+      // we need to scroll to where the node would be rendered
+      const left = coordinates.depth * 16;
+      scrollContainer?.scrollBy({
+        left,
+      });
+    }
+  }, []);
```

Comment:
If you move this outside the scope of `ViewHierarchy`, I think you might not need to use `useCallback`? Since it wont be recreated on each rerender of `ViewHierarchy` and it also doesn't seem to depend on anything in ViewHierarchy

Reply:
Done!